### ISSUE TITLE
Implement IM

### ISSUE THREAD

**Issue by joker314** [Collar-Flower/Collar-Flower#1]

**Comment by CosmicWebServices** [comment]
OK

**Comment by CosmicWebServices** [comment]
So first how do we have people invite people

**Comment by joker314** [comment]
We'll need to send the chat through somewhere, perhaps through Scratch comments: this might sound quite difficult -- but a user types the ID of a project to collab. Both people work on the same project, but if a non-collaborator enters the ID, a comment is dropped in

**Comment by joker314** [comment]
This comment gets sent to all collaborators through the IM system.

So basically, all coms are via comments

**Comment by CosmicWebServices** [comment]
Ah we also get the benifit of scratches whitelist/blacklist and we do not need to maintain comments

**Comment by CosmicWebServices** [comment]
OK so design wise how will this work?

**Comment by joker314** [comment]
File menu => Select project from web => Pop up asking for URL => Fetch username of author, feth collaborators (from comments), compare => Comment on project wityh either "Request to join" or "Joined" depending on if they are a collaborator or not.

**Comment by CosmicWebServices** [comment]
OK cool

**Comment by joker314** [comment]
Any subsequent comments they make will start with # to signify comment vs metadata

**Comment by CosmicWebServices** [comment]
yep

**Comment by CosmicWebServices** [comment]
Wait what project will all of these comments happen? The projects they are collaborating on...

**Comment by CosmicWebServices** [comment]
Also the projects NEED to be shared?

**Comment by joker314** [comment]
No, projects can be unshared, you can still comment on them with API

**Comment by CosmicWebServices** [comment]
How about for collaborations you are asking for the URL...

**Comment by joker314** [comment]
URL of the project you are collabing on.

**Comment by CosmicWebServices** [comment]
yeah...

**Comment by joker314** [comment]
Then they use comments to track diffs and changes, then the owner can merge what they have collabed on to the URL they linked to. Comments are deleted once all team members have seen it / time expired

**Comment by CosmicWebServices** [comment]
ah I see

**Comment by CosmicWebServices** [comment]
Should we start with this?
